source/middlewares/extract-jwt.ts

```
@@ -0,0 +1,35 @@
+import Jwt from 'jsonwebtoken';
+import { Context } from 'koa';
+import Config from '../configs/config';
+
+
+export async function  verifyToken(
+  ctx: Context,
+  next
+) {
+
+  const bearer = ctx.get('Authorization').split(' ');
+
+  let allowed = true;
+  let decoded;
+
+  const token = bearer[1];
+
+  if (bearer[0] !== 'Bearer') {
+    allowed = false;
+  }
+
+  if (!token) {
+    allowed = false;
```

You can throw right away if `token` is not provided.